casanovo/data/db_utils.py

```diff
@@ -0,0 +1,353 @@
+                & (self.db_peptides["calc_mass"] <= upper_bound)
+            ]
+            candidates.append(window[["peptide", "calc_mass", "protein"]])
+
+        candidates = pd.concat(candidates)
+        candidates.drop_duplicates(inplace=True)
+        candidates.sort_values(by=["calc_mass", "peptide"], inplace=True)
+        return candidates["peptide"], candidates["protein"]
+
+    def get_associated_protein(self, peptide: str) -> str:
+        """
+        Returns the associated protein for a given peptide.
+
+        Parameters
+        ----------
+        peptide : str
+            The peptide sequence.
+
+        Returns
+        -------
+        protein : str
+            The associated protein.
+        """
+        return self.db_peptides[self.db_peptides["peptide"] == peptide][
+            "protein"
+        ].values[0]
+
+    def _digest_fasta(
+        self,
+        fasta_filename: str,
+        enzyme: str,
+        digestion: str,
+        missed_cleavages: int,
+        max_mods: int,
+        min_peptide_length: int,
+        max_peptide_length: int,
+    ) -> pd.DataFrame:
+        """
+        Digests a FASTA file and returns the peptides, their masses, and associated protein.
+
+        Parameters
+        ----------
+        fasta_filename : str
+            Path to the FASTA file.
+        enzyme : str
+            The enzyme to use for digestion.
+            See pyteomics.parser.expasy_rules for valid enzymes.
+        digestion : str
+            The type of digestion to perform. Either 'full' or 'partial'.
+        missed_cleavages : int
+            The number of missed cleavages to allow.
+        max_mods : int
+            The maximum number of modifications to allow per peptide.
+        min_peptide_length : int
+            The minimum length of peptides to consider.
+        max_peptide_length : int
+            The maximum length of peptides to consider.
+
+        Returns
+        -------
+        mod_peptide_list : pd.DataFrame
+            A Pandas DataFrame with peptide, mass,
+            and protein columns. Sorted by neutral mass in ascending order.
+        """
+        # Verify the existence of the file:
+        if not os.path.isfile(fasta_filename):
+            logger.error("File %s does not exist.", fasta_filename)
+            raise FileNotFoundError(f"File {fasta_filename} does not exist.")
+
+        peptide_list = []
+        if digestion not in ["full", "partial"]:
+            logger.error("Digestion type %s not recognized.", digestion)
+            raise ValueError(f"Digestion type {digestion} not recognized.")
+        if enzyme not in parser.expasy_rules:
+            logger.info(
+                "Enzyme %s not recognized. Interpreting as cleavage rule.",
+                enzyme,
+            )
+        semi = digestion == "partial"
+        valid_aa = set(
+            [re.sub(r"[^A-Z]+", "", res) for res in self.residues.keys()]
```

Comment:
Changed to: valid_aa = set(list(self.residues.keys()) + ["C"])

Since C has a constant modification in the casanovo vocabulary it is always paired with a +57 in the residues dict. I needed to manually add it in plainly so it is considered a valid AA